psst-gui/src/webapi/client.rs

```diff
@@ -38,6 +39,43 @@ use crate::{
 
 use super::{cache::WebApiCache, local::LocalTrackManager};
 
+#[derive(Debug, Clone, Data, Lens, Deserialize)]
+pub struct TrackCredits {
+    #[serde(rename = "trackUri")]
+    pub track_uri: String,
+    #[serde(rename = "trackTitle")]
+    pub track_title: String,
+    #[serde(rename = "roleCredits")]
+    pub role_credits: Arc<Vec<RoleCredit>>,
+    #[serde(rename = "extendedCredits")]
+    pub extended_credits: Arc<Vec<String>>,
+    #[serde(rename = "sourceNames")]
+    pub source_names: Arc<Vec<String>>,
+}
+
+#[derive(Debug, Clone, Data, Lens, Deserialize)]
+pub struct RoleCredit {
+    #[serde(rename = "roleTitle")]
+    pub role_title: String,
+    pub artists: Arc<Vec<CreditArtist>>,
+}
+
+#[derive(Debug, Clone, Data, Lens, Deserialize)]
+pub struct CreditArtist {
```

Comment:
This could be in the lyrics file, which could then be imported into this file. This is what is usually done.